tests/envs/test_registration.py

```diff
@@ -23,12 +25,111 @@ def __init__(self, arg1, arg2, arg3):
 )
 
 
+@pytest.fixture(scope="function")
+def register_some_envs():
+    namespace = "MyAwesomeNamespace"
+    name = "MyAwesomeEnv"
+    gym.register(
+        id=namespace + "/" + name + "-v3",
+        entry_point="tests.envs.test_registration:ArgumentEnv",
+        kwargs={
+            "arg1": "arg1",
+            "arg2": "arg2",
+            "arg3": "arg3",
+        },
+    )
+    gym.register(
+        id=namespace + "/" + name + "-v1",
+        entry_point="tests.envs.test_registration:ArgumentEnv",
+        kwargs={
+            "arg1": "arg1",
+            "arg2": "arg2",
+            "arg3": "arg3",
+        },
+    )
+    yield
+    del gym.envs.registry.env_specs.tree[namespace][name]
+
+
 def test_make():
     env = envs.make("CartPole-v0")
     assert env.spec.id == "CartPole-v0"
     assert isinstance(env.unwrapped, cartpole.CartPoleEnv)
 
 
+@pytest.mark.parametrize(
+    "env_id, namespace, name, version",
+    [
+        (
+            "MyAwesomeNamespace/MyAwesomeEnv-v0",
+            "MyAwesomeNamespace",
+            "MyAwesomeEnv",
+            0,
+        ),
+        ("MyAwesomeEnv-v0", None, "MyAwesomeEnv", 0),
+        ("MyAwesomeEnv", None, "MyAwesomeEnv", None),
+        ("MyAwesomeEnv-vfinal-v0", None, "MyAwesomeEnv-vfinal", 0),
+        ("MyAwesomeEnv-vfinal", None, "MyAwesomeEnv-vfinal", None),
+        ("MyAwesomeEnv--", None, "MyAwesomeEnv--", None),
+        ("MyAwesomeEnv-v", None, "MyAwesomeEnv-v", None),
+    ],
+)
+def test_register(env_id, namespace, name, version):
+    envs.register(env_id)
+    assert gym.envs.spec(env_id).id == env_id
+    assert version in gym.envs.registry.env_specs.tree[namespace][name].keys()
+    del gym.envs.registry.env_specs.tree[namespace][name]
```

Comment:
Should probably `del gym.envs.registry.env_specs[env_id]` so we properly run `__delitem__` and we cleanup any empty leftovers.

Reply:
Yes, good catch!